### PR TITLE
critique.py was not working

### PR DESCRIPTION
**Pull request by prjemian**
this gets it working again

### CODE REVIEW

**Comment by prjemian** [comment]
Make this work with Python3, while we are at it?  Why not?

**Comment by prjemian** [comment]
`os.path.walk` is deprecated, switch to `os.walk` and refactor

File: critique.py [hunk 1]
```diff
@@ -149,11 +181,12 @@ def get_default(group, subgroupclass):
         if self.openHDF5(fullname):
             entry = get_default(self.hdf5, 'NXentry')
             if entry:
-                data = get_default(entry, 'NXdata')
-                if data:
-                    if 'signal' in data.attrs:
-                        signalname = data.attrs['signal']
-                        compliance = signalname in data
+                compliance = True
+                # data = get_default(entry, 'NXdata')
```
Comment (PeterC-DLS) [inline critique.py]: Remove commented out code and replace with comment. Also it seems that you return None

File: critique.py [hunk 2]
```diff
@@ -197,13 +234,22 @@ def walk_function(registrar, path, files):
...
+    print("# Critique of *exampledata* files")
+    print("date: %s" % datetime.datetime.now())
     registrar.report()
-    # TODO: should modify the README.rst
-    #    after the line that reads:
-    #    .. --- CRITIQUE report starts after this line ---
 
 
 if __name__ == '__main__':
+    # ./critique.py | tee critique.md
```
Comment (PeterC-DLS) [inline critique.py]: Remove

File: critique.py [hunk 3]
```diff
@@ -67,129 +83,37 @@ def __init__(self, path, fname):
...
-        except IOError:
-            return False
-        return True    
+        return self.describe_file() 
     
-    def find_NX_class_nodes(self, parent = None, nx_class = 'NXentry'):
+    def find_NX_class_nodes(self, parent, nx_class = 'NXentry'):
         '''identify the NXentry (or as specified) nodes'''
-        parent = parent or self.hdf5
+        parent = parent
```
Comment (PeterC-DLS) [inline critique.py]: Delete

**Comment by prjemian** [comment]
Note that file `./DLS/i16/538039.nxs` is reported as `HDF5 file` but not recognized as NeXus.  This is due to the underlying *h5py* package installed during analysis (version 2.9.0) that does not recognize the utf8 strings in the attributes.  Update of h5py to 2.10.0 produces the desired result.

**Pull request merged**